tracer/test/Datadog.Trace.TestHelpers.AutoInstrumentation/TestHelper.cs

```diff
@@ -93,9 +93,10 @@ public async Task<Process> StartDotnetTestSample(MockTracerAgent agent, string a
             }
 
             Output.WriteLine($"Starting Application: {sampleAppPath} {arguments ?? string.Empty}");
-            string testCli = forceVsTestParam ? EnvironmentHelper.GetDotnetExe() : EnvironmentHelper.GetDotNetTest();
+            string testCli = forceVsTestParam || useDotnetExec ? EnvironmentHelper.GetDotnetExe() : EnvironmentHelper.GetDotNetTest();
             string exec = testCli;
-            string appPath = testCli.StartsWith("dotnet") || testCli.Contains("dotnet.exe") || forceVsTestParam ? $"vstest {sampleAppPath}" : sampleAppPath;
+            string appPath = useDotnetExec ?  $"exec {sampleAppPath}" :
+                             testCli.StartsWith("dotnet") || testCli.Contains("dotnet.exe") || forceVsTestParam ? $"vstest {sampleAppPath}" : sampleAppPath;
```

Comment:
nit:
```suggestion
            bool usesVsTest = testCli.StartsWith("dotnet") || testCli.Contains("dotnet.exe") || forceVsTestParam;
            string appPath = (useDotnetExec, usesVsTest) switch
            {
                (true, _) => $"exec {sampleAppPath}",
                (_, true) => $"vstest {sampleAppPath}",
                _ => sampleAppPath,
            };
```

Reply:
done in https://github.com/DataDog/dd-trace-dotnet/pull/6573/commits/0bb1ed8bf288df45f34491c03669193dd5f59b8b